google-cloud-firestore/samples/query_watch.rb

```diff
@@ -0,0 +1,199 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+
+def listen_document project_id:, collection_path: "cities"
+  # project_id = "Your Google Cloud Project ID"
+  # collection_path = "cities"
+
+  firestore = Google::Cloud::Firestore.new project_id: project_id
+  # [START fs_listen_document]
+  doc_ref = firestore.col(collection_path).doc "SF"
+  snapshots = []
+
+  # Watch the document.
+  listener = doc_ref.listen do |snapshot|
+    # Process the snapshot.
+    puts "Received document snapshot: #{snapshot.document_id}"
+    snapshots << snapshot
+  end
+  # [END fs_listen_document]
+
+  sleep 1
```

Comment:
Rather than sleeping, you should filter out non-existing snapshots in the listener.

Reply:
So it would be acceptable to show the filtering in the published sample? (The `sleep` statements are outside of the `fs_listen_document` region tags demarcating the sample.)

Reply:
We should show this filtering, because it gives an important insight into how to use the API.

Reply:
Understood, I will add it. Thanks!

Reply:
Removed sleep after relaxing criteria of `wait_until`.